src/screens/SearchScreen.js

```diff
@@ -9,19 +9,21 @@ import {initializeMap, updateViewStyle, mapLoaded, updateRegion} from "../action
 
 import _ from 'lodash';
 
-import {CardSection, Card} from '../components/common/index';
-
-import {map} from '../constants';
+import {map, navKeys} from '../constants';
 
 
 class SearchScreen extends Component {
     componentWillMount() {
+        const {token, navigation: {navigate, state: {key}}} = this.props;
+
+        if (!token && key !== navKeys.LOGIN) {
+            navigate(navKeys.LOGIN);
```

Comment:
Same pattern we've seen a few times; if we don't have a token, go back to `LOGIN`.